tests/ovm_test.go

```diff
@@ -0,0 +1,347 @@
+
+			currOVM = newOfflineVirtualMachine(false)
+
+			// Start and stop VM multiple times
+			for i := 0; i < 5; i++ {
+				By(fmt.Sprintf("Doing run: %d", i))
+				startOVM(currOVM)
+				stopOVM(currOVM)
+			}
+		})
+
+		It("should not update the VM spec if Running", func() {
+			newOVM := newOfflineVirtualMachine(true)
+
+			Eventually(func() bool {
+				newOVM, err = virtClient.OfflineVirtualMachine(newOVM.Namespace).Get(newOVM.Name, &v12.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return hasCondition(newOVM, v1.OfflineVirtualMachineRunning)
+			}, 360*time.Second, 1*time.Second).Should(BeTrue())
+
+			By("Updating the OVM template spec")
+			newOVM, err = virtClient.OfflineVirtualMachine(newOVM.Namespace).Get(newOVM.Name, &v12.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			updatedOVM := newOVM.DeepCopy()
+			updatedOVM.Spec.Template.Spec.Domain.Resources.Requests = v13.ResourceList{
+				v13.ResourceMemory: resource.MustParse("4096Ki"),
+			}
+			updatedOVM, err := virtClient.OfflineVirtualMachine(updatedOVM.Namespace).Update(updatedOVM)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Expecting the old VM spec still running")
+			vm, err := virtClient.VM(newOVM.Namespace).Get(newOVM.Name, v12.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			vmMemory := vm.Spec.Domain.Resources.Requests.Memory()
+			ovmMemory := newOVM.Spec.Template.Spec.Domain.Resources.Requests.Memory()
+			Expect(vmMemory.Cmp(*ovmMemory)).To(Equal(0))
+
+			By("Restarting the OVM")
+			newOVM = stopOVM(newOVM)
+			newOVM = startOVM(newOVM)
+
+			By("Expecting updated spec running")
+			vm, err = virtClient.VM(newOVM.Namespace).Get(newOVM.Name, v12.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			vmMemory = vm.Spec.Domain.Resources.Requests.Memory()
+			ovmMemory = updatedOVM.Spec.Template.Spec.Domain.Resources.Requests.Memory()
+			Expect(vmMemory.Cmp(*ovmMemory)).To(Equal(0))
+		})
+
+		It("should survive guest shutdown, multiple times", func() {
+			By("Creating new OVM, not running")
+			newOVM := newOfflineVirtualMachine(false)
+			newOVM = startOVM(newOVM)
+			var vm *v1.VirtualMachine
+			var err error
+
+			for i := 0; i < 3; i++ {
+				By("Getting the running VM")
+				Eventually(func() bool {
+					vm, err = virtClient.VM(newOVM.Namespace).Get(newOVM.Name, v12.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					return vm.Status.Phase == v1.Running
+				}, 240*time.Second, 1*time.Second).Should(BeTrue())
+
+				By("Obtaining the serial console")
+				expecter, _, err := tests.NewConsoleExpecter(virtClient, vm, "serial0", 10*time.Second)
+				Expect(err).ToNot(HaveOccurred())
+				defer expecter.Close()
+
+				By("Guest shutdown")
+				_, err = expecter.ExpectBatch([]expect.Batcher{
+					&expect.BSnd{S: "\n"},
+					&expect.BExp{R: "login as 'cirros' user. default password: 'gocubsgo'. use 'sudo' for root."},
+					&expect.BSnd{S: "\n"},
+					&expect.BExp{R: "cirros login:"},
```

Comment:
We will need to change the login code here once #775 is merged. The `cirros login:` line will the contain the vm name and the login fails. Will comment back here once #775 is merged.

Reply:
#775 is merged. You will need to update this section to:

```golang
expecter, err := tests.LoggedInCirrosExpecter(vm)
Expect(err).ToNot(HaveOccurred())
defer expecter.Close()
res, err := expecter.ExpectBatch([]expect.Batcher{
	&expect.BSnd{S: "sudo poweroff\n"},
	&expect.BExp{R: "The system is going down NOW!"},
}, time.Second*10)
```

to successfully login to cirros.